src/uk/gov/hmcts/contino/YarnBuilder.groovy

```diff
@@ -303,6 +305,19 @@ EOF
     return status == 0  // only a 0 return status is success
   }
 
+  private auditDisparity() {
+    def status = steps.sh label: "Determine whether yarn audit returns transitive dependency issues that are not currently caught.",
+      script: "yarn npm audit --recursive --environment production", returnStatus: true
+    return status
+  }
+
+  private nagAboutYarnAuditChange() {
+    if (auditDisparity()){
+      WarningCollector.addPipelineWarning("transitive_dependency_audit_incoming",
+        "You are receiving this warning as you will be directly impacted by this change as there are existing vulnerabilities in your transitive dependencies. We will be adding the --recursive flag to CVE reporting. This means that your transitive dependencies within frontend repositories will also be checked for vulnerabilities. Run `yarn npm audit --recursive --environment production` ahead of time to catch and fix issues so you won't be blocked on the day of the change.", LocalDate.of(2023, 03, 15))
+    }
```

Comment:
please indent properly.
if block close } and function } shouldn't be at the same level